include/CL/opencl.hpp

```diff
@@ -4586,6 +4613,73 @@ class Image : public Memory
         }
         return param;
     }
+
+#if defined(cl_ext_image_requirements_info)
+    static cl_int getRequirementsInfoExt(const Context &context,
+        cl_mem_flags flags,
+        ImageRequirementsInfoExt param_name,
+        size_type param_value_size,
+        void* param_value,
+        size_type* param_value_size_ret,
+        const cl_mem_properties* properties = nullptr,
+        const ImageFormat* image_format = nullptr,
+        const cl_image_desc* image_desc = nullptr)
+    {
+#if CL_HPP_TARGET_OPENCL_VERSION >= 120
+        Device device = context.getInfo<CL_CONTEXT_DEVICES>().at(0);
+        cl_platform_id platform = device.getInfo<CL_DEVICE_PLATFORM>();
+        CL_HPP_INIT_CL_EXT_FCN_PTR_PLATFORM_(platform, clGetImageRequirementsInfoEXT);
+#else
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clGetImageRequirementsInfoEXT);
+#endif
+        if (pfn_clGetImageRequirementsInfoEXT == nullptr) {
+            return detail::errHandler(CL_INVALID_OPERATION, __GET_IMAGE_REQUIREMENT_INFO_EXT_ERR);
+        }
+
+        return detail::errHandler(pfn_clGetImageRequirementsInfoEXT(context(),
+                properties,
+                flags,
+                image_format,
+                image_desc,
+                static_cast<cl_image_requirements_info_ext>(param_name),
+                param_value_size,
+                param_value,
+                param_value_size_ret),
+                __GET_IMAGE_REQUIREMENT_INFO_EXT_ERR);
+    }
+
+static cl_int getRequirementsInfoExt(const Context &context,
+        cl_mem_flags flags,
+        ImageRequirementsInfoExt param_name,
+        size_type param_value_size,
+        void* param_value,
+        size_type* param_value_size_ret,
```

Comment:
Sorry I didn't notice this before: It'd be best to avoid having these kinds of param_value_size, param_value, and param_value_size_ret parameters in the C++ bindings.  Can we do something more like this instead?

https://github.com/KhronosGroup/OpenCL-CLHPP/pull/225#discussion_r1166271210

Basically, we'd need one variant that is templated on the info being queried, and another version that takes the info as an argument.  In both cases we shouldn't need void pointers, though.

Reply:
@bashbaug  I am agree, best would be to use detail::getInfo method, but still there are a few more args:
image_format, image_desc, (properties) which should be forwarded also to the getInfoHelper in a proper order.

To solve this, I need to define own version of getInfoHelper function.
What do you think ?

Reply:
Please have a look at the wrapper around `clGetKernelWorkGroupInfo` - it also needs to pass an additional device argument.  I think you could do something similar for this wrapper?

Reply:
Yes I saw this, but still there wasn't this what I would expected.
I create helper method: getImageRequirementsInfoExtHelper